src/core/thread/network_data_leader_ftd.cpp

```diff
@@ -479,7 +484,8 @@ otError Leader::RlocLookup(uint16_t aRloc16, bool &aIn, bool &aStable, uint8_t *
                     {
                         borderRouterEntry = borderRouter->GetEntry(i);
 
-                        if (borderRouterEntry->GetRloc() == aRloc16)
+                        if ((aExactMatch && borderRouterEntry->GetRloc() == aRloc16) ||
+                            (!aExactMatch && (borderRouterEntry->GetRloc() & (~Mle::kMaxChildId)) == aRloc16))
```

Comment:
How about something like `GetRouterId(borderRouterEntry->GetRloc()) == routerId)`?  And compute `routerId` once at the beginning?

Reply:
Thanks for raising this. I am thinking routerId computation may not be required if aExactMatch is true. 
How about define a static function like `RouterIdMatch(RlocA, RlocB)` to hidden the bit operating here?

Reply:
Submitted one patch which uses the RouterIdMatch() for your review temporarily.
I am still open to update as your suggestion if you think it would be better.